File: app/src/main/java/com/example/noahdoperalskigreenflag/MainActivity.java
package com.example.noahdoperalskigreenflag;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    Button createButton;

    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        getSupportActionBar().hide();
        createButton = findViewById(R.id.btnCreateAccount);
        createButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view)
            {
                Intent myIntent = new Intent(MainActivity.this, SecondActivity.class);
                startActivity(myIntent);
            }
        });
    }
}

File: app/src/main/java/com/example/mvpexample/view/MainActivity.kt
package com.example.mvpexample.view

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.recyclerview.widget.RecyclerView
import com.example.mvpexample.R
import com.example.mvpexample.model.Movie
import com.example.mvpexample.presenter.MoviePresenter

class MainActivity : AppCompatActivity() , MainView{

    private lateinit var moviePresenter: MoviePresenter
    private lateinit var recyclerView: RecyclerView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    override fun showProgress() {

        //TODO("Not yet implemented")
    }

    override fun hideProgress() {

        //TODO("Not yet implemented")
    }

    override fun setData(movies: List<Movie>) {

        //TODO("Not yet implemented")
    }

    override fun setError(msg: String) {

    }
}

// interfaces
// contract between the view layer and presenter layer
interface MainView {
    fun showProgress()
    fun hideProgress()
    fun setData(movies: List<Movie>)
    fun setError(msg: String)

}